ci/cudf_pandas_scripts/run_tests.sh

```diff
@@ -75,7 +75,7 @@ IFS=',' read -r -a versions <<< "$output"
 
 for version in "${versions[@]}"; do
     echo "Installing pandas version: ${version}"
-    python -m pip install "pandas==${version}"
+    python -m pip install "numpy>=1.23,<2.0a0" "pandas==${version}"
```

Comment:
In the rerun got the original failure with incompatible Pandas & NumPy versions as explained above ( https://github.com/rapidsai/cudf/pull/16300#issuecomment-2307889745 ). That said, now see this is due to PR ( https://github.com/rapidsai/cudf/pull/16595 ), which added legacy Pandas testing

Wrote much more details about this in the commit message ( https://github.com/rapidsai/cudf/pull/16300/commits/3bf2311bf19abe6372ca3320b74908bfdf6091e7 ), the gist is [only Pandas 2.2.2+ is NumPy 2 compatible]( https://pandas.pydata.org/pandas-docs/version/2.2.2/whatsnew/v2.2.2.html#pandas-2-2-2-is-now-compatible-with-numpy-2-0 ). However when `pip` downgrades Pandas here NumPy is not downgraded from NumPy 2 to NumPy 1. As these older Pandas versions don't have upper bounds (restricting to NumPy 1), `pip` won't downgrade them (though Idk whether that would have been enough for `pip` anyways)

To fix this issue, am just pinning to NumPy 1 in this install command. Have confirmed this works locally and expect it to work here as well

Since we already test with NumPy 2 above (not to mention in various other tests here), think those tests are sufficient and it is fine to stick to NumPy 1 for these older Pandas versions (even as the long tail includes more NumPy 2 support over time)

https://github.com/rapidsai/cudf/blob/3bf2311bf19abe6372ca3320b74908bfdf6091e7/ci/cudf_pandas_scripts/run_tests.sh#L64-L69

Reply:
Confirmed this fixed [that job]( https://github.com/rapidsai/cudf/actions/runs/10533694328/job/29192863387?pr=16300 )